bftengine/tests/testSeqNumForClientRequest/seqNumForClientRequest_test.cpp

```diff
@@ -1,8 +1,80 @@
 #include "gtest/gtest.h"
+#include "SimpleClientImp.cpp"
 
-TEST(seqNumForClientRequest_test, initial_test) {}
+using namespace std;
+using namespace bftEngine;
 
-int main(int argc, char* argv[]) {
+namespace {
+
+class mockSeqNumGenerator : public SeqNumberGeneratorForClientRequestsImp {
+ public:
+  uint32_t getLastCountOfUniqueFetchID() { return lastCountOfUniqueFetchID_; }
+
+  void setLastCountOfUniqueFetchID(uint64_t count) { lastCountOfUniqueFetchID_ = count; }
+};
+
+TEST(seqNumForClientRequest_test, send_request_fast_still_persistence) {
+  SeqNumberGeneratorForClientRequests *seqNumGen = new mockSeqNumGenerator();
+
+  set<u_int64_t> seqNums;
+
+  for (int i = 0; i < 10; i++) {
+    seqNums.insert(seqNumGen->generateUniqueSequenceNumberForRequest());
+  }
+
+  Assert(seqNums.size() == 10);
```

Comment:
What does this assert tell you? Can you maybe add comments on each assert, in order to explain what property/invariant you are testing?

Reply:
Sure.